### PR TITLE
Refinement of the dialog box

### PR DESCRIPTION
**Pull request by seryal**
Refinement of the dialog box.
Before:

After:

### CODE REVIEW

File: widgets/dialogs.pas [hunk 1]
```diff
@@ -136,21 +136,28 @@   TMessageDialog = class(TWForm)
...
     const
       CControlsSpacing = 2;
-      CMinDialogHeight = 200;
+      CMinDialogHeight = 150;
       CMinDialogWidth = 300;
       CMinButtonHeight = 25;
       CMinButtonWidth = 100;
       CMinImageHeight = 70;
       CMinImageWidth = 70;
+      CTitileHeight = 24;
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: This should be `CTitleHeight`, right? 😉

File: widgets/dialogs.pas [hunk 2]
```diff
@@ -278,6 +287,33 @@ procedure TMessageDialog.KeyDown(var Key: NativeInt; Shift: TShiftState);
...
 end;
 
+procedure TMessageDialog.Show;
+var
+  ownForm: TWForm;
+  curHeight: Integer;
+begin
+  inherited Show;
+  if not (Owner is TWForm) then
+    exit;
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: `Exit`, `Break` and `Continue` should be written as Identifiers in PascalCase (cause in fact they _are_ identifiers, one can overload them with e.g. variables 😋)

File: widgets/dialogs.pas [hunk 3]
```diff
@@ -278,6 +287,33 @@ procedure TMessageDialog.KeyDown(var Key: NativeInt; Shift: TShiftState);
...
+  ownForm := TWForm(Owner);
```
Comment (pascaldragon) [inline widgets/dialogs.pas:298]: I think that doesn't belong here considering you had done that further down

File: widgets/dialogs.pas [hunk 4]
```diff
@@ -278,6 +287,33 @@ procedure TMessageDialog.KeyDown(var Key: NativeInt; Shift: TShiftState);
...
+    curHeight := FMessageText.HandleElement.scrollHeight + CMinButtonHeight + CControlsSpacing * 8 + CTitileHeight;
+    if curHeight - CMinImageWidth > ownForm.Height then
+      curHeight := ownForm.Height - CMinImageWidth;
+    if curHeight < CMinDialogHeight then
+      curHeight := CMinDialogHeight;
+    Self.Height := curHeight;
+    fMessagePanel.Height := FMessageText.HandleElement.scrollHeight;
+    if Owner is TWForm then begin
+      ownForm := TWForm(Owner);
+      Self.Top := round(OwnForm.Height/2 - Self.Height/2);
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: Style corrections:
- `Round` in PascalCase
- spaces between operators (thus `Self.Height / 2`)
- I know that Pascal is case insensitive, but please use the casing you used for the variable, here `ownForm`

File: widgets/dialogs.pas [hunk 5]
```diff
@@ -296,6 +332,31 @@ constructor TMessageDialog.Create(AOwner: TComponent);
...
+
+    fTitlePanel := TWPanel.Create(Self);
+    fTitlePanel.BeginUpdate;
+    try
+      fTitlePanel.Parent := Self;
+      fTitlePanel.BorderSpacing.Around := CControlsSpacing;
+      fTitlePanel.Height := CTitileHeight;
+      fTitlePanel.BevelOuter := bvNone;
+      fTitlePanel.Align := alTop;
+      fTitlePanel.Color:= clSkyBlue;
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: Missing space before the assignment operator

File: widgets/dialogs.pas [hunk 6]
```diff
@@ -296,6 +332,31 @@ constructor TMessageDialog.Create(AOwner: TComponent);
...
+    finally
+      fTitlePanel.EndUpdate;
+    end;
+    fTitleText := TWLabel.Create(Self);
+    fTitleText.BeginUpdate;
+    try
+      fTitleText.Parent := fTitlePanel;
+      fTitleText.BorderSpacing.Around := CControlsSpacing;
+      fTitleText.Align := alClient;
+      fTitleText.Alignment:= taCenter;
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: Again a missing space

File: widgets/dialogs.pas [hunk 7]
```diff
@@ -307,10 +368,20 @@ constructor TMessageDialog.Create(AOwner: TComponent);
...
       FInfoImage.EndUpdate;
     end;
-    FMessageText := TWLabel.Create(Self);
+    fMessagePanel := TWPanel.Create(Self);
+    fMessagePanel.BeginUpdate;
+    try
+      fMessagePanel.Parent := Self;
+      fMessagePanel.BorderSpacing.Around := CControlsSpacing;
+      fMessagePanel.BevelOuter := bvNone;
+      fMessagePanel.Align:=alClient;
```
Comment (pascaldragon) [inline widgets/dialogs.pas]: And again spaces

**Comment by pascaldragon** [comment]
Thank you very much. I might be critizing every small detail I come across as you can see by my review, but I very much appreciate that you help here. 😊 If you fix the above issues I'll merge it.

File: widgets/dialogs.pas (hunk 1 above)
Comment (seryal) [reply widgets/dialogs.pas]: Oh, yes. Title :) Fixed.

File: widgets/dialogs.pas (hunk 2 above)
Comment (seryal) [reply widgets/dialogs.pas]: Fixed.

File: widgets/dialogs.pas (hunk 3 above)
Comment (seryal) [reply widgets/dialogs.pas:298]: Thanks, at the time of this writing I got distracted and the code was duplicated. 
Fixed.

File: widgets/dialogs.pas (hunk 4 above)
Comment (seryal) [reply widgets/dialogs.pas]: Fixed.

File: widgets/dialogs.pas (hunk 5 above)
Comment (seryal) [reply widgets/dialogs.pas]: Fixed.

File: widgets/dialogs.pas (hunk 6 above)
Comment (seryal) [reply widgets/dialogs.pas]: Fixed.

File: widgets/dialogs.pas (hunk 7 above)
Comment (seryal) [reply widgets/dialogs.pas]: Fixed.

**Comment by seryal** [comment]
> Thank you very much. I might be critizing every small detail I come across as you can see by my review, but I very much appreciate that you help here. 😊 If you fix the above issues I'll merge it.

No problem, this is a very useful skill for me. :) 
But sorry that you have to lose time pointing out my errors :(

**Comment by pascaldragon** [comment]
Thank you for the corrections and the contribution as a whole! 😊

**Pull request merged**